### PR TITLE
Add tracking for Zskera Vaults collectibles

### PR DESCRIPTION
**Pull request by cyriun**

### CODE REVIEW

**Comment by cyriun** [comment]
quick draft of a possible way to add these pets, i think it works, ill do more testing after i sleep haha, gimme feedback

**Comment by Duckwhale** [comment]
Looks reasonable to me, all things considered (since we don't have a proper method for objects like that yet).

The conflicts are likely from merging in the other PRs, but should be easily resolved :)

**Comment by cyriun** [comment]
yeah its pretty much me moving to new stuff before current stuff gets added and writing over the same lines

**Comment by cyriun** [comment]
added a toy thats obtained from the same chest list as the pets but wondering if theres an easier way to add it then what i did?

**Comment by Duckwhale** [comment]
Uhm, you can do something like what I did here: https://github.com/WowRarity/Rarity/pull/609/files It's still a bit messy, but that's just how things are currently. Using `addAttemptForItem` saves some repetition, but it needs a bit of streamlining (eventually...).

**Comment by cyriun** [comment]
forgot to update comment to say "pets and toys" oh well

File: Core/EventHandlers.lua [hunk 1]
```diff
@@ -1637,6 +1637,48 @@ function R:OnEvent(event, ...)
...
+			["Shattered Crystals"] = true,
+			["Chest of Ice"] = true,
+			["Forgotten Lockbox"] = true,
+			["Slimy Goo"] = true,
+			["Wind-Bound Strongbox"] = true,
+			["Spun Webs"] = true,
+			["Obsidian Grand Cache"] = true,
+			["Frozen Coffer"] = true,
+		}
+		if Rarity.isFishing and Rarity.isOpening and Rarity.lastNode and vaultChests[Rarity.lastNode] then
```
Comment (Duckwhale) [inline Core/EventHandlers.lua:1672]: Should this not be using the localized version? (prefixed by `L`)

At least that's what all the other code is doing. For example:

```
-- Handle opening Timeless Chest
	if Rarity.isFishing and Rarity.isOpening and Rarity.lastNode and (Rarity.lastNode == L["Timeless Chest"]) then
	-- <the rest of the code>
```

**Comment by Duckwhale** [comment]
This still needs a rebase, but is it otherwise complete?

**Comment by cyriun** [comment]
yeah if it all looks good to you all the chests and items have been added

**Pull request merged**